flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientHandlerTest.java

```diff
@@ -244,54 +237,89 @@ public void testCancelBeforeActive() throws Exception {
 	}
 
 	/**
-	 * Verifies that {@link RemoteInputChannel} is enqueued in the pipeline, and
-	 * {@link AddCredit} message is sent to the producer.
+	 * Verifies that {@link RemoteInputChannel} is enqueued in the pipeline for notifying credits,
+	 * and verifies the behaviour of credit notification by triggering channel's writability changed.
 	 */
 	@Test
 	public void testNotifyCreditAvailable() throws Exception {
+		final SingleInputGate inputGate = createSingleInputGate();
+		final RemoteInputChannel inputChannel1 = spy(createRemoteInputChannel(inputGate));
+		final RemoteInputChannel inputChannel2 = spy(createRemoteInputChannel(inputGate));
 		final CreditBasedClientHandler handler = new CreditBasedClientHandler();
-		final EmbeddedChannel channel = new EmbeddedChannel(handler);
+		final EmbeddedChannel channel = spy(new EmbeddedChannel(handler));
```

Comment:
I think this `spy` is not needed anywhere?

Reply:
agree